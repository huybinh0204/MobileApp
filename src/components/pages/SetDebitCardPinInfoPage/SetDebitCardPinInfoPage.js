import { useNavigation, useRoute } from '@react-navigation/native';
import React from 'react';
import { IconSvg } from '_components/atoms';
import { SecondaryScreenLayout } from '_components/organisms';
import { ICONS, NAVIGATION } from '_constants';
import strings from '_localization';
import { normalize } from '_utilities/screen';
import { Container, Description, SetPinButton, Title } from './SetDebitCardPinInfoPage.styles';

const SetDebitCardPinInfoPage = () => {
  const { params } = useRoute();
  const { navigate, popToTop, goBack } = useNavigation();

  const { isChangePinVariant } = params;
  const content = isChangePinVariant ? strings.card.changePinInfo : strings.card.setInitialPinInfo;

  const setDebitCardPin = () => {
    navigate(NAVIGATION.card.setDebitCardPinForm, { isChangePinVariant });
  };

  const handleBackPress = () => {
    if (isChangePinVariant) {
      goBack();
    } else {
      popToTop();
    }
  };
  return (
    <SecondaryScreenLayout
      testID="SetDebitCardPinInfoPage"
      onBackPress={handleBackPress}
      title={isChangePinVariant ? content.header : null}
    >
      <Container>
        <IconSvg width={normalize(250)} height={normalize(250)} icon={ICONS.setDebitCardPin} />
        <Title>{content.title}</Title>
        <Description>{content.description}</Description>
        <SetPinButton accessibilityLabel="Set Debit Card Pin" onPress={setDebitCardPin}>
          {content.openForm}
        </SetPinButton>
      </Container>
    </SecondaryScreenLayout>
  );
};

export default SetDebitCardPinInfoPage;
