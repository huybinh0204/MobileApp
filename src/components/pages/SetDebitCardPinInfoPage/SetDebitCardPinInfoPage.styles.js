import styled from '@emotion/native';
import { BaseText, MainButton } from '_components/atoms';

export const Container = styled.View(({ theme }) => ({
  flex: 1,
  alignItems: 'center',
  padding: theme.spacing.m,
}));

export const Title = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading4,
  color: theme.colors.beta900,
  marginVertical: theme.spacing.s,
  textAlign: 'center',
}));

export const Description = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta500,
  paddingHorizontal: theme.spacing.m,
  textAlign: 'center',
}));

export const SetPinButton = styled(MainButton)({
  marginTop: 'auto',
});
