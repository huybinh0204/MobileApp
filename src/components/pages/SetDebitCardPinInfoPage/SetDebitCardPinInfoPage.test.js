import { useNavigation } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { NAVIGATION } from '_constants';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import SetDebitCardPinInfoPage from './SetDebitCardPinInfoPage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

describe('SetDebitCardPinInfoPage', () => {
  const mockedNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useNavigation.mockReturnValue(mockedNavigation);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<ComponentWithProviders component={SetDebitCardPinInfoPage} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate to the set pin form', async () => {
    const { getByAccessibilityLabel } = render(
      <ComponentWithProviders
        component={SetDebitCardPinInfoPage}
        params={{ isChangePinVariant: false }}
      />
    );

    await waitFor(() => {
      fireEvent.press(getByAccessibilityLabel('Set Debit Card Pin'));
    });

    expect(mockedNavigation.navigate).toBeCalledWith(NAVIGATION.card.setDebitCardPinForm, {
      isChangePinVariant: false,
    });
  });

  it('should goBack when isChangePinVariant is true and back button is pressed', async () => {
    const { getByTestId } = render(
      <ComponentWithProviders
        component={SetDebitCardPinInfoPage}
        params={{ isChangePinVariant: true }}
      />
    );

    const backButton = getByTestId('BackButton');
    await fireEvent.press(backButton);
    expect(mockedNavigation.goBack).toBeCalled();
  });

  it('should popToTop when isChangePinVariant is false and back button is pressed', async () => {
    const { getByTestId } = render(
      <ComponentWithProviders
        component={SetDebitCardPinInfoPage}
        params={{ isChangePinVariant: false }}
      />
    );

    const backButton = getByTestId('BackButton');
    await fireEvent.press(backButton);

    expect(mockedNavigation.popToTop).toBeCalled();
  });
});
