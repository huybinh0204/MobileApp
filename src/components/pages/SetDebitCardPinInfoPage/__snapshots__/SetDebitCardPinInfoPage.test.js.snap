// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetDebitCardPinInfoPage should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="SetDebitCardPinInfoPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <SvgMock
                              height={453}
                              width={453}
                            />
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 43.5,
                                    "lineHeight": 60,
                                    "marginVertical": 29,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Letâ€™s set your PIN
                            </Text>
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#5F6A77",
                                    "fontFamily": "Poppins",
                                    "fontSize": 25.5,
                                    "lineHeight": 35.5,
                                    "paddingHorizontal": 58,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Set your card PIN number to start using your Kinly Card.
                            </Text>
                            <View
                              accessibilityLabel="Set Debit Card Pin"
                              accessibilityState={
                                Object {
                                  "disabled": false,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignSelf": "stretch",
                                  "backgroundColor": "#3D3397",
                                  "borderRadius": 10,
                                  "marginTop": "auto",
                                  "opacity": 1,
                                  "paddingHorizontal": 45.5,
                                  "paddingVertical": 31,
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                disabled={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Set my card PIN
                              </Text>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
